### PR TITLE
Add non compliant and compliant samples for Go detectors

### PR DESCRIPTION
**Pull request by ankit-amazon**
#### Add non compliant and compliant samples for Go detectors. 
Note: All the test cases have 100% recall and precision.

### CODE REVIEW

File: src/rule_based_detector_library/GO/detectors/go-hidden-goroutine/go-hidden-goroutine_compliant.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	"fmt"
+)
+
+func hiddenGoroutineCompliant() {
+	// Compliant: There are other operations along with the goroutine.
+	if num := 17; num < 0 {
+		fmt.Println(num, "is negative")
+	}
+	go func() {
+		if 5%2 == 0 {
```
Comment (bhaaayus) [inline src/rule_based_detector_library/GO/detectors/go-hidden-goroutine/go-hidden-goroutine_compliant.go:19]: `%` should have one space before and after for better readability

File: src/rule_based_detector_library/GO/detectors/go-hidden-goroutine/go-hidden-goroutine_compliant.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+			fmt.Println("5 is even")
+		} else {
+			fmt.Println("5 is odd")
+		}
+	}()
+	if 8%2 == 0 {
```
Comment (bhaaayus) [inline src/rule_based_detector_library/GO/detectors/go-hidden-goroutine/go-hidden-goroutine_compliant.go:25]: `%` should have one space before and after for better readability. Please incorporate these changes for all the files.

File: src/rule_based_detector_library/GO/detectors/go-http-listenandservetls/go-http-listenandservetls_compliant.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ */
+
+package httputil
+
+// {fact rule=go-http-listenandservetls@v1.0 defects=0}
+import (
+	"net/http"
+)
+
+func httpListenandservetlsCompliant(certFile string, keyFile string) {
```
Comment (bhaaayus) [inline src/rule_based_detector_library/GO/detectors/go-http-listenandservetls/go-http-listenandservetls_compliant.go:13]: Should be `httpListenAndServeTLSCompliant`

File: src/rule_based_detector_library/GO/detectors/go-http-listenandservetls/go-http-listenandservetls_non-compliant.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ */
+
+package httputil
+
+// {fact rule=go-http-listenandservetls@v1.0 defects=1}
+import (
+	"net/http"
+)
+
+func httpListenandservetlsNoncompliant() {
```
Comment (bhaaayus) [inline src/rule_based_detector_library/GO/detectors/go-http-listenandservetls/go-http-listenandservetls_non-compliant.go:13]: Should be `httpListenAndServeTLSNoncompliant`

**Review by bhaaayus — CHANGES_REQUESTED** [review]
Please address the comments

**Review by bhaaayus — APPROVED** [review]

**Review by atonaamz — APPROVED** [review]

**Review by atonaamz — DISMISSED** [review]

**Comment by atonaamz** [comment]
Can we have the same move these samples to `go/src/detectors`?

**Review by atonaamz — APPROVED** [review]

**Pull request merged**